### PR TITLE
Add  translations-WIP

### PR DESCRIPTION
**Pull request by lukefromdc**
This is the start of translation support. WIP for now as more is needed. We need the translations themselves (for the .desktop file) and a way to copy MATE.desktop.in with the translations included over the untranslated MATE.desktop file which can then be removed from the source files.

I do not know if the gsettings override file needs to be translated.

### CODE REVIEW

**Comment by raveit65** [comment]
> I do not know if the gsettings override file needs to be translated.

I don't think, ihmo. For fedora i created an untranslated override file. https://src.fedoraproject.org/rpms/mate-desktop/blob/rawhide/f/mate-rhel.gschema.override
Here only the key name is used which is untranslated in when i open them in dconf-editor.

File: po/LINGUAS [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# please keep this list sorted alphabetically
+#
+af
```
Comment (raveit65) [inline po/LINGUAS:3]: Should be without locale. If a language is translated at transifex ( pull it with `tx pull -af`) you will see a *.po file in that directory. Then you can add the locale name to LINGUAS.
Here you mark which language translations (*.po files) are available for this project.

**Comment by lukefromdc** [comment]
tx pull -af gets command not found, I don't have transifex installed nor an account there,

**Comment by lukefromdc** [comment]
I was able to install the binary for transifex easily enough, but tx pull doesn't work without an account. An account created from my github account was able to get through the initial steps, but I don't understand enough about it to get `tx pull` to work. I also see the account is a 15 day trial, I do not have any form of online currency, have no paid accounts anywhere, and that is not about to change. Someone else will have to deal with that unless they can tell me exactly how to get tx pull to work once only.

**Comment by raveit65** [comment]
I checked this with another github account of mine. I only got this trial account where i can buy an expensive subscription to add a new project with own collaborators 0_o
No idea, what's going on with transifex and why old projects like mate-desktop are free.
Looks like transifex is a commercial company nowadays.

**Comment by raveit65** [comment]
Back in my normal transifex account i saw 2 join request to mate-desktop project wich i did with my new account.
grafik
So it seems that it is possible to add people with this `trial account`.
I am thinking this is free.

**Comment by lukefromdc** [comment]
I was unable even to connect to the existing MATE project on Transifex, coming up at mate-2 instead

**Comment by lukefromdc** [comment]
This is what I got, note that I had to block a number of 3ed party JS providers that are never allowed on any of my machines, notably googletagmanager

Transifex

**Comment by raveit65** [comment]
Try this site https://www.transifex.com/solutions/for-translators/ and choose join a project. Than search for mate-desktop and click on Mate button. 
grafik
On next side click on join project. I will add you tomorrow. Now it's very late an d i go to bed.
Or try it now.

**Comment by raveit65** [comment]
Today i had to spend more than 5 hours to fix broken translation strings before releasing.
I complete frustrated about this translation stuff and i will never do it again in life.
Should another person takes the hand over Mate at transifex.
This is complete out of control and translators do always mistakes.
Some languages, ie. jews are currently complete broken in every repo.
Honestly, i don't mind........ and if a new updated language breaks building i simply do `git restore po/shiti.po`  :P
Not my task any more.

**Comment by lukefromdc** [comment]
The session should be identifiable without being translated, we are dealing with only one file here. The name "MATE" stays the same in the x11 session's .desktop file for every language using the same or similar alphabets English does. I propose to leave this open for someone else to deal with (it's a draft after all) and we then have the option of releasing the session untranslated as it will still work and is considered experimental at this time. 

If a release is untenable I can bump the version number and be done with that for the moment.

**Comment by lukefromdc** [comment]
I managed to create the .pot file by hand

**Comment by raveit65** [comment]
`./autogen.sh && ./makepot` just creates mate-session-wayland.pot and move it to top dir, same as in other packages.
But the naming of the package is really confusing. In configure.ac it called `AC_INIT([mate-session-wayland],`.
But the repo called mate-wayland-session. I am thinking this is one reason why `distcheck doesn't work in other PR.
IHMO, you should rename the repository or naming in configure and where ever it is writen.

When this is done and PR is merged i can the package to transifex.

File: .tx/config [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[main]
+host =  https://www.transifex.com
+
+[o:mate:p:MATE:r:master--mate-wayland-session]
+file_filter  = po/<lang>.po
+source_file  = mate-wayland-session.pot
```
Comment (raveit65) [inline .tx/config:6]: Also, here the name should match naming in configure.ac and repository name.

File: po/POTFILES.skip [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+data/firedecor.config
+data/mate/close.png
+data/mate/close-hovered.png
+data/mate/close-pressed.png
+data/mate/minimize.png
+data/mate/minimize-hovered.png
+data/mate/minimize-pressed.png
+data/mate/toggle-maximize.png
+data/mate/toggle-maximize-hovered.png
+data/mate/toggle-maximize-pressed.png
```
Comment (raveit65) [inline po/POTFILES.skip:11]: Ihmo, you don't need the graphic files. But check how it is handle in other repositories.

File: mate-session-wayland.pot [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR MATE Desktop Environment team
+# This file is distributed under the same license as the mate-session-wayland package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: mate-session-wayland 1.27.0\n"
+"Report-Msgid-Bugs-To: https://github.com/mate-desktop/mate-session-wayland/"
```
Comment (raveit65) [inline mate-session-wayland.pot:9]: As you can see here the naming is wrong of the repository in this generated file.
This comes from AC_INIT in configure.
So naming needs updated.

**Comment by lukefromdc** [comment]
I will rename the package in configure.ac and anywhere else it is wrong

**Comment by lukefromdc** [comment]
Package name should now be consistant everywhere

File: configure.ac [hunk 5]
```diff
@@ -16,14 +16,21 @@ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
...
 PKG_PROG_PKG_CONFIG([0.19])
 
-# Workaround to make aclocal get the right flags
-AC_SUBST(ACLOCAL_AMFLAGS, "\${ACLOCAL_FLAGS}")
+# ****************************
+AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.8])
+
+GETTEXT_PACKAGE=mate-session-wayland
```
Comment (raveit65) [inline configure.ac:24]: This needs to be renamed too after renaming ACinit.

File: makepot [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#! /bin/sh
+
+PACKAGE=mate-session-wayland;
+
```
Comment (raveit65) [inline makepot:4]: same here

**Comment by raveit65** [comment]
***After*** renaming everything to mate-wayland-session you can drop mate-session-wayland.pot. Then you can create the source file again with `./autogen.sh && ./makepot`

File: configure.ac (hunk 5 above)
Comment (lukefromdc) [reply configure.ac:24]: That came back in when merging your work with mine, fixing it now

File: makepot (hunk 6 above)
Comment (lukefromdc) [reply makepot:4]: Found this one too, fixing now

**Comment by lukefromdc** [comment]
`tx pull` still got
```
MATE.master--mate-wayland-session - Resource MATE.master--mate-wayland-session does not exist
[##############################] (1 / 1)
luke@ubuntu:~/Desktop/mate-wayland-session$ 
```
possibly as this is not on master

**Comment by raveit65** [comment]
> `tx pull` still got
> 
> ```
> MATE.master--mate-wayland-session - Resource MATE.master--mate-wayland-session does not exist
> [##############################] (1 / 1)
> luke@ubuntu:~/Desktop/mate-wayland-session$ 
> ```
> 
> possibly as this is not on master

Yes, 
.......did you add  the package to transifex on their website?

**Comment by lukefromdc** [comment]
I would not know how to determine if they are needed or not, will remove them at your suggestion though, assuming if not listed in other packages not needed here

**Comment by lukefromdc** [comment]
I don't think it would be possible to add the project to transifex without first 
merging this PR (so they would not be stuck on the branch),
 but I really don't know and so far have yet to find the interface for doing so

**Comment by lukefromdc** [comment]
Image files removed from POTFILES.skip at your suggestion

**Comment by raveit65** [comment]
> I don't think it would be possible to add the project to transifex without first merging this PR (so they would not be stuck on the branch), but I really don't know and so far have yet to find the interface for doing so.

Of course fist it needs to be in master. I only asked because it wasn't clear for me, after you ask for the `tx pull -af` command.
I will change PR from draft to review.

**Review by raveit65 — APPROVED** [review]
LGTM

**Comment by raveit65** [comment]
I think this is OK for now.
We need some translated strings from transifex for further testing.

offtopic on
distcheck fails on this branch because of missing PNGs in distcheck
```
mkdir -p /media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-wayland-session-dist/mate-wayland-session-1.27.0/_inst/share/firedecor/button-styles/mate/
cp close.png /media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-wayland-session-dist/mate-wayland-session-1.27.0/_inst/share/firedecor/button-styles/mate/close.png
cp: cannot stat 'close.png': No such file or directory
make[4]: *** [Makefile:437: install-data-local] Error 1
```
offtopic off

**Comment by raveit65** [comment]
`tx pull -af` works like a charm.
```
<cut>
MATE.master--mate-wayland-session [yo] - Minimum translation completion threshold not satisfied, skipping
MATE.master--mate-wayland-session [zh] - Minimum translation completion threshold not satisfied, skipping
MATE.master--mate-wayland-session [zh-Hans] - Minimum translation completion threshold not satisfied, skipping
MATE.master--mate-wayland-session [zh_CN] - Done
MATE.master--mate-wayland-session [zh_HK] - Minimum translation completion threshold not satisfied, skipping
MATE.master--mate-wayland-session [zh_TW] - Done
MATE.master--mate-wayland-session [zu] - Minimum translation completion threshold not satisfied, skipping
[##############################] (176 / 176)
[rave@mother mate-wayland-session]$ git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
(use "git add <file>..." to include in what will be committed)
po/ca.po
po/cs.po
po/da.po
po/de.po
po/en_GB.po
po/es.po
po/fr.po
po/gl.po
po/hu.po
po/id.po
po/it.po
po/ja.po
po/lt.po
po/ms.po
po/nb.po
po/nl.po
po/oc.po
po/pl.po
po/pt.po
po/pt_BR.po
po/ru.po
po/sl.po
po/sr.po
po/tr.po
po/uk.po
po/zh_CN.po
po/zh_TW.po

nothing added to commit but untracked files present (use "git add" to track)
[rave@mother mate-wayland-session]$

After you did this for yourself add those po files with `git add ......` , add them to LINGUAS and commit.

**Comment by raveit65** [comment]
I did not translate the keywords `Keywords=launch;MATE;desktop;session;` in German language, because i do not understand how this works and if this really needs to be translated.

**Comment by raveit65** [comment]
Not done, something is missing ;-)
I wrote it already here 
```
After you did this for yourself add those po files with git add ...... , add them to LINGUAS and commit.
```
Without this important step, *.po files will never used during compiling.
```

The day will come that you have to do this without my help....

**Comment by lukefromdc** [comment]
I can get on that one now, thanks

**Comment by lukefromdc** [comment]
https://github.com/mate-desktop/mate-wayland-session/commit/7b9a86abe28523272cb11c6d01328b28bf9b66ec
added the .po files but didn't add them to Linguas, so last push hopefully fixed that

**Comment by raveit65** [comment]
Thanks, you got it.
This is important do when a new language comes in with a pull from transifex

**Comment by lukefromdc** [comment]
One remaining problem:
https://github.com/mate-desktop/mate-wayland-session/issues/8
translations not built with meson. I should be able to use another packages po/meson.build as a template to fix that

**Pull request merged**